controllers/cluster_controller.go

```diff
@@ -247,7 +247,7 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 				continue
 			}
 
-			if !accessor.GetDeletionTimestamp().IsZero() {
+			if accessor.GetDeletionTimestamp() != nil && !accessor.GetDeletionTimestamp().IsZero() {
```

Comment:
Technically don't need a nil check here, as a nil metav1.Time returns the proper true value for IsZero

Reply:
Ditto for the other similar changes

Reply:
That's true for typed objects, but when using unstructured https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/unstructured.go#L360

Reply:
It's the same for unstructured (don't need a nil check) unless I'm missing something? https://github.com/kubernetes/apimachinery/blob/2373d029717c4d169463414a6127cd1d0d12680e/pkg/apis/meta/v1/time.go#L61-L63

Reply:
The metav1.Time type behaves properly, it's the `GetDeletionTimestamp` method that returns nil https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/unstructured.go#L363-L365

Reply:
Still fine: it return's a `(*metav1.Time)(nil)`: https://play.golang.org/p/3G463erHe4z